docs/source/gbd2019_models/hemoglobin_anemia_and_iron_deficiency/hemoglobin/index.rst

```diff
@@ -6,4 +6,237 @@ Hemoglobin Model
+  mgumbel_mv2p = function(mn, vr){
+    list(
+      alpha = XMAX - mn - EULERS_CONSTANT*sqrt(vr)*sqrt(6)/pi,
+      scale = sqrt(vr)*sqrt(6)/pi)}
+  # function to randomly sample n times from mirror gumbel distribution
+  rmgumbel = function(n, alpha, scale){
+    mn = alpha + scale*EULERS_CONSTANT
+    rgumbel(n, alpha+XMAX-(2*mn), scale)}
+
+  # function to sample from ensemble hemoglobin distribution using functions defined above
+    # n = number of samples to take
+    # mn = mean hemoglobin concentration
+    # vr = hemoglobin distribution variance (standard deviation squared)
+    # w = list of ensemble distribution weights c(gamma_weight, mirror_gumbel_weight)
+  sample_from_hemoglobin_distribution <- function(n, mn, vr, w){
+
+    ##parameters
+    params_gamma = gamma_mv2p(mn, vr)
+    params_mgumbel = mgumbel_mv2p(mn, vr)
+
+    ##weighting
+    sample_list = sum(
+      w[1] * rgamma(n, params_gamma$shape, params_gamma$rate), 
+      w[2] * rmgumbel(n, params_mgumbel$alpha, params_mgumbel$scale)
+    )
+    sample_list
+  }
+
+  # function to calculate area under curve below threshold q for mirror gumbel distribution
+  pmgumbel = function(q, alpha, scale, lower.tail){ 
+    #NOTE: with mirroring, take the other tail
+    pgumbel(XMAX-q, alpha, scale, lower.tail=ifelse(lower.tail,FALSE,TRUE))}
+
+  # function to calculate area under curve of hemoglobin ensemble distribution using the functions defined above
+    # q = hemoglobin threshold
+    # mn = mean hemoglobin concentration
+    # vr = hemoglobin variance (standard deviation squared)
+    # w = list of ensemble distribution weights c(gamma_weight, mirror_gumbel_weight)
+  ens_mv2prev <- function(q, mn, vr, w){
+    x = q
+
+    ##parameters
+    params_gamma = gamma_mv2p(mn, vr)
+    params_mgumbel = mgumbel_mv2p(mn, vr)
+
+    ##weighting
+    prev = sum(
+      w[1] * pgamma(x, params_gamma$shape, params_gamma$rate), 
+      w[2] * pmgumbel(x, params_mgumbel$alpha, params_mgumbel$scale, lower.tail=T))
+    prev}
+
+Pregnancy Adjustment
+^^^^^^^^^^^^^^^^^^^^
+
+To sample hemoglobin values for pregnant/lactating women, use the same functions as above, but multiply the hemoglobin mean and standard deviation parameters used for those functions by the respective pregnancy adjustment factors listed below. Notably, the GBD 2019 assumes that the pregnancy adjustment factor applies to 40 weeks of gestation and 6 weeks post-gestation.
+
+.. list-table:: Pregnancy Adjustment Factors
+  :widths: 15, 30, 10
+  :header-rows: 1
+
+  * - Parameter
+    - Value
+    - Note
+  * - Mean hemoglobin adjustment factor
+    - 0.919325
+    - No uncertainty is used in the GBD 2019 code
+  * - Hemoglobin standard deviation adjustment factor
+    - 1.032920188
+    - No uncertainty is used in the GBD 2019 code
+
+Validation Criteria
++++++++++++++++++++
+
+Hemoglobin samples should satisfy the following criteria:
+
+- all_samples > 0
+- all_samples < XMAX = 220
```

Comment:
Technically, I don't think these inequalities follow from the definition of the hemoglobin distribution. The [support](https://en.wikipedia.org/wiki/Support_(mathematics)) (i.e. the range of x values that occur with positive probability) of the gamma distribution is [0,infinity), and the support of the (mirror) gumbel distribution is (-infinity, infinity), so theoretically *any* x value is possible, similar to a normal distribution. Presumably, however, the probability of seeing a value outside the range [0,220] is very small...

Reply:
As you mentioned yesterday @alibow , having a negative hemoglobin level doesn't make any sense, so maybe reframe these as checks that should be built into the model by the software engineers: "If a sample is negative, resample!"

Reply:
Agreed! I worked this in.